src/test/java/com/jcabi/log/VerboseProcessTest.java

```diff
@@ -236,14 +244,163 @@ public void logsErrorWhenUnderlyingStreamIsClosed() throws Exception {
     }
 
     /**
-     * VerboseProcess calls destroy() on the underlying Process when it is
+     * VerboseProcess can terminate its monitors and underlying Process when
      * closed.
      * @throws Exception If something goes wrong
      */
     @Test
-    public void destroysUnderlyingProcessWhenClosed() throws Exception {
+    public void terminatesMonitorsAndUnderlyingProcessWhenClosed()
+        throws Exception {
+        final InputStream inputStream = new InfiniteInputStream('i');
+        final InputStream errorStream = new InfiniteInputStream('e');
         final Process prc = Mockito.mock(Process.class);
-        new VerboseProcess(prc).close();
-        Mockito.verify(prc, Mockito.atLeastOnce()).destroy();
+        Mockito.doReturn(inputStream).when(prc).getInputStream();
+        Mockito.doReturn(errorStream).when(prc).getErrorStream();
+        Mockito.doAnswer(
+            new Answer<Void>() {
+                @Override
+                public Void answer(final InvocationOnMock invocation)
+                    throws Exception {
+                    inputStream.close();
+                    errorStream.close();
+                    return null;
+                }
+            }
+        ).when(prc).destroy();
+        final VerboseProcess verboseProcess = new VerboseProcess(
+            prc,
+            Level.FINEST,
+            Level.FINEST
+        );
+        final StringWriter writer = new StringWriter();
+        final WriterAppender appender = new WriterAppender(
+            new SimpleLayout(),
+            writer
+        );
+        appender.addFilter(new VrbPrcMonitorFilter(verboseProcess));
+        org.apache.log4j.Logger.getLogger(
+            VerboseProcess.class
+        ).addAppender(appender);
+        final int readInterval = 50;
+        new Timer(true).schedule(
+            new TimerTask() {
+                @Override
+                public void run() {
+                    verboseProcess.close();
+                }
+            },
+            readInterval
+        );
+        verboseProcess.stdoutQuietly();
+        final int monitorsInterval = 500;
```

Comment:
you can use `com.jcabi.aspects.Tv` for numeric constants, and get rid of this variable